ethereum/jsonrpc/src/main/java/tech/pegasys/pantheon/ethereum/jsonrpc/websocket/subscription/blockheaders/NewBlockHeadersSubscriptionService.java

```diff
@@ -38,20 +38,28 @@ public NewBlockHeadersSubscriptionService(
 
   @Override
   public void onBlockAdded(final BlockAddedEvent event, final Blockchain blockchain) {
-    final List<NewBlockHeadersSubscription> subscribers =
-        subscriptionManager.subscriptionsOfType(
-            SubscriptionType.NEW_BLOCK_HEADERS, NewBlockHeadersSubscription.class);
+    subscriptionManager
+        .getVertx()
+        .<Void>executeBlocking(
+            future -> {
+              final List<NewBlockHeadersSubscription> subscribers =
+                  subscriptionManager.subscriptionsOfType(
+                      SubscriptionType.NEW_BLOCK_HEADERS, NewBlockHeadersSubscription.class);
 
-    final Hash newBlockHash = event.getBlock().getHash();
+              final Hash newBlockHash = event.getBlock().getHash();
 
-    for (final NewBlockHeadersSubscription subscription : subscribers) {
-      final BlockResult newBlock =
-          subscription.getIncludeTransactions()
-              ? blockWithCompleteTransaction(newBlockHash)
-              : blockWithTransactionHash(newBlockHash);
+              for (final NewBlockHeadersSubscription subscription : subscribers) {
+                final BlockResult newBlock =
+                    subscription.getIncludeTransactions()
+                        ? blockWithCompleteTransaction(newBlockHash)
+                        : blockWithTransactionHash(newBlockHash);
 
-      subscriptionManager.sendMessage(subscription.getId(), newBlock);
-    }
+                subscriptionManager.sendMessage(subscription.getId(), newBlock);
+              }
+
+              future.complete();
+            },
+            result -> {});
```

Comment:
If an exception is thrown from the action, does it wind up being logged or does this result handle wind up silently ignoring it?

Reply:
Currently ignored, will add logging.